From ba9cd51ab422fe457bdac37c9797ae869c1893da Mon Sep 17 00:00:00 2001
From: Meng Li <Meng.Li@windriver.com>
Date: Fri, 20 Feb 2026 08:24:13 -0800
Subject: [PATCH 59/59] sound: soc: replace of_property_read_bool() with
 of_property_present()

When system boots up, there are lots of warning output as below:
OF: <dts node>: Read of boolean property <property> with a value.
This issue is introduced by upstream commit c141ecc3cecd("of:Warn
when of_property_read_bool() is used on non-boolean properties"), so it
needs to replace of_property_read_bool() with of_property_present().

Upstream-Status: Pending

Signed-off-by: Meng Li <Meng.Li@windriver.com>
Signed-off-by: Matt Madison <matt@madison.systems>
---
 sound/soc/tegra/tegra_asoc_machine.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/sound/soc/tegra/tegra_asoc_machine.c b/sound/soc/tegra/tegra_asoc_machine.c
index 94daf793..99772b08 100644
--- a/sound/soc/tegra/tegra_asoc_machine.c
+++ b/sound/soc/tegra/tegra_asoc_machine.c
@@ -58,7 +58,7 @@ static bool of_dai_link_is_available(struct device_node *link_node)
 		 * number of links that DT exposes). Other codec subnodes
 		 * can be empty and populated in override file.
 		 */
-		if (!of_property_read_bool(child, DAI) &&
+		if (!of_property_present(child, DAI) &&
 		    !of_node_cmp(child->name, "codec"))
 			continue;
 
@@ -89,7 +89,7 @@ static int of_get_child_count_with_name(struct device_node *node,
 
 	for_each_child_of_node(node, child)
 		if (!of_node_cmp(child->name, name) &&
-		    of_property_read_bool(child, DAI))
+		    of_property_present(child, DAI))
 			num++;
 
 	return num;
@@ -478,7 +478,7 @@ static int parse_dt_dai_links(struct snd_soc_card *card,
 			if (of_node_cmp(codec->name, "codec"))
 				continue;
 
-			if (!of_property_read_bool(codec, DAI)) {
+			if (!of_property_present(codec, DAI)) {
 				dev_dbg(&pdev->dev,
 					"sound-dai prop missing for (%pOF)\n",
 					codec);
@@ -615,7 +615,7 @@ int parse_card_info(struct snd_soc_card *card, struct snd_soc_ops *pcm_ops,
 		return ret;
 
 	/* parse machine DAPM widgets */
-	if (of_property_read_bool(node, PREFIX "widgets")) {
+	if (of_property_present(node, PREFIX "widgets")) {
 		ret = snd_soc_of_parse_audio_simple_widgets(card,
 			PREFIX "widgets");
 		if (ret < 0)
@@ -628,7 +628,7 @@ int parse_card_info(struct snd_soc_card *card, struct snd_soc_ops *pcm_ops,
 	 * which require them to be connected to machine source/sink
 	 * DAPM widgets.
 	 */
-	if (of_property_read_bool(node, PREFIX "routing")) {
+	if (of_property_present(node, PREFIX "routing")) {
 		ret = snd_soc_of_parse_audio_routing(card, PREFIX "routing");
 		if (ret < 0)
 			return ret;
-- 
2.43.0

